File: src/2605.form-smallest-number-from-two-digit-arrays.2668/solution.ts
function minNumber (nums1: number[], nums2: number[]): number {
    const flags1 = new Array<boolean>(10);
    let min1 = 9;
    const flags2 = new Array<boolean>(10);
    let min2 = 9;
    for (const n of nums1) {
        flags1[n] = true;
        min1 = Math.min(n, min1);
    }
    for (const n of nums2) {
        flags2[n] = true;
        min2 = Math.min(n, min2);
    }
    for (let i = 1; i < 10; i++) {
        if (flags1[i] && flags2[i]) {
            return i;
        }
    }

    return Math.min(min1 * 10 + min2, min2 * 10 + min1);
}

File: src/2605.form-smallest-number-from-two-digit-arrays.2668/solution.java
class Solution {
    public int minNumber(int[] nums1, int[] nums2) {
        boolean[] exists1 = normalize(nums1);
        boolean[] exists2 = normalize(nums2);
        for(int i=1;i<10;i++){
            if(exists1[i] && exists2[i]){
                return i;
            }
        }
        int min1 = getMin(exists1);
        int min2 = getMin(exists2);
        if(min1<min2){
            return min1*10+min2;
        }
        
        return min2*10+min1;
    }
    
    private int getMin(boolean[] exists){
        int result = 0;
        for(int i=0;i< exists.length;i++){
            if(exists[i]){
                result = i;
                break;
            }
        }
        return result;
    }
    
    private boolean[] normalize(int[] nums){
        boolean[] result = new boolean[10];
        for(int n:nums){
            result[n] = true;
        }
        return result;
    }
}